src/combinations.rs

```diff
@@ -77,6 +77,21 @@ impl<I: Iterator> Combinations<I> {
             self.pool.prefill(k);
         }
     }
+
+    /// For a given size `n`, return the count of remaining elements or None if it would overflow.
+    fn remaining_for(&self, n: usize) -> Option<usize> {
```

Comment:
Should this really be a method of `Combinations` and accept `n` (as `n` is given by `Combinations::n`)?

Reply:
Well `Combinations::size_hint` also calls it on `low` and `upp` so not just `Combinations::n`.